src/sentry/api/endpoints/event_grouping_info.py

```diff
@@ -39,9 +42,26 @@ def get(self, request: Request, project, event_id) -> Response:
         hashes = event.get_hashes()
 
         try:
-            variants = event.get_grouping_variants(
-                force_config=config_name, normalize_stacktraces=True
-            )
+            if event.get_event_type() == "transaction":
+                # Transactions events are grouped using performance detection. They
+                # are not subject to grouping configs, and the only relevant
+                # grouping variant is `PerformanceProblemVariant`.
+
+                problems = EventPerformanceProblem.fetch_multi([(event, h) for h in hashes.hashes])
+
+                # Create a variant for every problem associated with the event
+                # TODO: Generate more unique keys, in case this event has more than
+                # one problem of a given type
+                variants = {
+                    problem.problem.type.name: PerformanceProblemVariant(problem)
+                    for problem in problems
+                    if problem
+                }
```

Comment:
This logic probably should be moved to `get_grouping_variants` at some point

Reply:
Makes sense!